### ISSUE TITLE
[request] Make it possible to lookup an agent by hostname

### ISSUE THREAD

**Issue by rossengeorgiev** [wazuh/wazuh-api#24]
When I want to grab agent details, or key, I need to know the agent id. It would be helpful if all that information can be looked up by hostname instead for the purpose of automated registration using something like salt, puppet, etc.

Currently, you have to workaround that by getting a list of all the agents, looking up the hostname to get the agent id, and then getting agent details, or key.

**Comment by mgmacias95** [comment]
Hello @rossengeorgiev,

Do you mean a call like this one?
```shellsession
# curl -u foo:bar -k "https://localhost:55000/agents/myNewAgent?pretty"
{
   "error": 0,
   "data": {
      "status": "Never connected",
      "ip": "any",
      "dateAdd": "2017-09-21 15:58:52",
      "id": "006",
      "name": "myNewAgent"
   }
}
```
This can be easily done and can be very helpful 😄 !

Best regards,
Marta

**Comment by rossengeorgiev** [comment]
@mgmacias95 Yeah, just like that.

**Comment by mgmacias95** [comment]
Hi again @rossengeorgiev,

We're working on this issue. I'd like to suggest you to use this API call while we finish the one you've requested:
```shellsession
# curl -u foo:bar "localhost:55000/agents?pretty&search=manager"
{
   "error": 0,
   "data": {
      "totalItems": 1,
      "items": [
         {
            "status": "Active",
            "name": "manager",
            "ip": "127.0.0.1",
            "version": "Wazuh v3.0.0-beta9",
            "os": {
               "platform": "centos",
               "version": "7",
               "name": "CentOS Linux"
            },
            "id": "000"
         }
      ]
   }
}
```
It is almost the same as the one you've requested but the results are saved inside the `items` list.

Best regards,
Marta

**Comment by rossengeorgiev** [comment]
Thanks for the workaround, but ideally need something with exact hostname matching.

**Comment by mgmacias95** [comment]
Hello @rossengeorgiev,

This request has just been implemented and is awaiting to be merged (see #42 ). It will be added in Wazuh 3.1. 

Best regards,
Marta
